Tests/SafeDIMacrosTests/InstantiableMacroTests.swift

```diff
@@ -1196,68 +1330,594 @@ import Testing
+						self.forwardedA = forwardedA
+						self.receivedA = receivedA
+					}
+
+					init(forwardedA: ForwardedA, receivedA: ReceivedA) {
+						self.forwardedA = forwardedA
+						self.receivedA = receivedA
+					}
+
+					public init(forwardedA: ForwardedA, receivedA: ReceivedA, receivedB: ReceivedB, customizable: String = "") {
+						self.forwardedA = forwardedA
+						self.receivedA = receivedA
+						self.receivedB = receivedB
+					}
+
+					let forwardedA: ForwardedA
+					let receivedA: ReceivedA
+					let receivedB: ReceivedB
+
+					public typealias ForwardedProperties = ForwardedA
+				}
+				"""
+			}
+		}
+
+		@Test
+		func declaration_fixit_updatesRequiredInitializerWhenDependencyMissingFromInitAndAccessibilityModifierMissing() {
+			assertMacro {
+				"""
+				@Instantiable
+				public struct ExampleService: Instantiable {
+					init(receivedA: ReceivedA, receivedB: ReceivedB, customizable: String = "") {
+						self.receivedA = receivedA
+						self.receivedB = receivedB
+					}
+
+					@Forwarded let forwardedA: ForwardedA
+					@Received let receivedA: ReceivedA
+					@Received let receivedB: ReceivedB
+				}
+				"""
+			} diagnostics: {
 				"""
 				@Instantiable
 				public struct ExampleService: Instantiable {
-				public init(
-				forwardedA: ForwardedA,
-				receivedA: ReceivedA,
-				receivedB: ReceivedB
-				) {
-				self.forwardedA = forwardedA
-				self.receivedA = receivedA
-				self.receivedB = receivedB
-				}
-
-				    public init(forwardedA: ForwardedA, receivedA: ReceivedA) {
-				        self.forwardedA = forwardedA
-				        self.receivedA = receivedA
-				        receivedB = ReceivedB()
-				    }
+					init(receivedA: ReceivedA, receivedB: ReceivedB, customizable: String = "") {
+				 ╰─ 🛑 @Instantiable-decorated type must have a `public` or `open` initializer with a parameter for each @Instantiated, @Received, or @Forwarded-decorated property.
+				    ✏️ Add arguments for forwardedA: ForwardedA
+						self.receivedA = receivedA
+						self.receivedB = receivedB
+					}
 
-				    @Forwarded let forwardedA: ForwardedA
-				    @Received let receivedA: ReceivedA
-				    @Received let receivedB: ReceivedB
+					@Forwarded let forwardedA: ForwardedA
+					@Received let receivedA: ReceivedA
+					@Received let receivedB: ReceivedB
 				}
 				"""
-			} expansion: {
+			} fixes: {
 				"""
+				@Instantiable
 				public struct ExampleService: Instantiable {
-				public init(
-				forwardedA: ForwardedA,
-				receivedA: ReceivedA,
-				receivedB: ReceivedB
-				) {
-				self.forwardedA = forwardedA
-				self.receivedA = receivedA
-				self.receivedB = receivedB
-				}
-
-				    public init(forwardedA: ForwardedA, receivedA: ReceivedA) {
-				        self.forwardedA = forwardedA
-				        self.receivedA = receivedA
-				        receivedB = ReceivedB()
-				    }
+					init(forwardedA: ForwardedA, receivedA: ReceivedA, receivedB: ReceivedB, customizable: String = "") {
```

Comment:
Should it also add `public` here?

Reply:
I guess will just do two passes on fix-its right now

Reply:
Ideally yes, but today we just does one fix at a time. There will be new fixit once the first fixit is applied.

Reply:
you beat me to it 😉